common/snapshotlog.py

```diff
@@ -38,18 +38,65 @@ class LogFilter(object):
                                     :py:data:`ERROR_AND_CHANGES`
         decode (encfstools.Decode): instance used for decoding lines or ``None``
     """
+
+    # TODO Better use an enumeration
     NO_FILTER =         0
     ERROR =             1
     CHANGES =           2
     INFORMATION =       3
     ERROR_AND_CHANGES = 4
+    RSYNC_TRANSFER_FAILURES = 5
 
+    # Regular expressions used for filtering log file lines.
+    # RegExp syntax see: https://docs.python.org/3.10/library/re.html#regular-expression-syntax
+    # (?:...) = the matched substring cannot be retrieved in a group (non-capturing)
     REGEX = {None:              None,
              NO_FILTER:         None,
              ERROR:             re.compile(r'^(?:\[E\]|[^\[])'),
              CHANGES:           re.compile(r'^(?:\[C\]|[^\[])'),
              INFORMATION:       re.compile(r'^(?:\[I\]|[^\[])'),
-             ERROR_AND_CHANGES: re.compile(r'^(?:\[E\]|\[C\]|[^\[])')}
+             ERROR_AND_CHANGES: re.compile(r'^(?:\[E\]|\[C\]|[^\[])'),
+             RSYNC_TRANSFER_FAILURES: re.compile(
```

Comment:
These strings are from the rsync sources?
Maybe can you add a link to the related file in the rsync repo here as a comment?

Reply:
Great idea (and lot of work ;-) but I will enrich this to make maintenance easier in the future.
Question is: Permalink to the examined rsync version or dynamic link (may change).
I guess the purpose is compare underlying source code of the implemented regexp against the most recent rsync version so I will add permalink.

Reply:
Not sure about permalink or dynamic link. The purpose for me would be the opportunity to compare the current latest/stable rsync version source with the strings we do use here.
If we stick to this solution than we can add a little helper script (as manual step in the release process) doing this checkup for us and alarm if the strings have modified.

Reply:
Oh, damn! I didn't mean commenting each one of this message. 🤣 
Just one single comment. Very nice!